packages/data-models/appConfig.ts

```diff
@@ -223,6 +240,44 @@ const APP_CONFIG = {
   TASKS,
 };
 
+/**
+ * Deprecation migrations to support renamed propertiesfor backwards compatibility
+ */
+const DEPRECATION_RULES = {
```

Comment:
nit(non-blocking): really nice to have the set of rules and clear authoring messages. I think there might be some minor issue if we need to change entirely either a root property name (e.g. `APP_FOOTER_DEFAULTS -> FOOTER`) or if we had more deeply nested structures. 

I'm guessing the solution would be flattening the entire config and deprecations so that you could define
```ts
{
  "APP_FOOTER_DEFAULTS.templateName":{
    newProperty: "APP_FOOTER_DEFAULTS.template"  
  }
}
```
however I'm also happy to just deal with this when the need arises